### ISSUE TITLE
Broken fragment links 

### ISSUE THREAD

**Issue by dstorey** [MicrosoftEdge/MicrosoftEdge-Documentation#13]
http://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/#setting-and-managing-breakpoints (line 37)

ID doesn’t have hyphens.

----
http://dev.modern.ie/platform/documentation/f12-devtools-guide/emulation

#mode and #display don’t exist. There is no named anchor above their sections.

-----

http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/#sending-info-to-the-console (line 12)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/#sending-javascript-into-the-console (line 11)

No named anchors. Probably should also remove hyphens like other IDs.

While on this page the "error codes" link misses its trailing / causing a re-direct. Other links to same page include it.

----

http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/console-error-and-status-codes/

Probably should remove actual anchor link to https://cdn.example.net as it is an example and doesn’t exist.

MathML link lacking its trailing /

----

http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/using-the-console-api/

One of the counters links is https://msdn.microsoft.com/en-us/library/dn706051(v=vs.85).aspx#counters rather than just #counters

All the following fragments fail (looks like no named anchors):

* Custom messages
* Inspectable objects and nodes
* Counters
* Timers
* Assertions
* Traces

----

http://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/webstorage-in-debugger/

broken fragment:
ttp://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/webstorage-in-debugger/#session-and-local-storage 

----

http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/

Broken fragments:
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#layout (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#color-picker (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#changes (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#events (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#computed (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#styles (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#pseudo-state (line 49)

**Comment by mattwojo** [comment]
Thanks @dstorey! I'll work my way through this list. 

Could you further explain:
http://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/#setting-and-managing-breakpoints (line 37)
ID doesn’t have hyphens.

*You are referring to this line?* 
The **Console** icon (next to the **Help** icon at the top of the F12 tools) or CTRL + ` opens the [**Console tool**](../console/) in a fourth pane below the *script* pane. Open it when you want to view **Console** output, or use the **Console** command line.

**Comment by mattwojo** [comment]
http://dev.modern.ie/platform/documentation/f12-devtools-guide/emulation
#mode and #display don’t exist. There is no named anchor above their sections.

*These do have named anchors and function properly in the GitHub readme, but not on the Modern.ie site... I think that markdown anchor links may not be set up to work on that site... @melanierichards can you confirm? Is this intentional?

**Comment by melanierichards** [comment]
Not sure, I think I heard about some problem with jump links before. @molant wrote the markdown parser so he would know about that.

**Comment by mattwojo** [comment]
Opened a new issue (https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/issues/14) for this anchor-linking issue.

**Comment by melanierichards** [comment]
Sounds good!

**Comment by mattwojo** [comment]
@dstorey regarding this comment: *While on this page the "error codes" link misses its trailing / causing a re-direct. Other links to same page include it.*

Where are you seeing this? 

Do you mean line 22 at https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/tree/master/f12-devtools-guide/console?  ...this link seems to function properly for me.

**Comment by mattwojo** [comment]
http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/console-error-and-status-codes/  *Probably should remove actual anchor link to https://cdn.example.net as it is an example and doesn’t exist.*   --> Done. (This was tricky as GitHub auto-links URLs in this format, gave it ticks to format it as code). 

*MathML link lacking its trailing /* --> @dstorey, I'm not sure what you mean by "trailing /" ...is this a standard format for Microsoft? These links seem to work fine for me.

**Comment by mattwojo** [comment]
http://dev.modern.ie/platform/documentation/f12-devtools-guide/console/using-the-console-api/

*One of the counters links is https://msdn.microsoft.com/en-us/library/dn706051(v=vs.85).aspx#counters rather than just #counters* --> FIXED (https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/commit/cb316336d2aebee38df7a373e19f44ab5f918214)

*All the following fragments fail (looks like no named anchors): Custom messages, Inspectable objects and nodes, Counters, Timers, Assertions, Traces*
---> All of the anchor issues will be addressed in the issue https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/issues/14

**Comment by mattwojo** [comment]
@dstorey -- I understand what this one means:

http://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/webstorage-in-debugger/

broken fragment:
ttp://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/webstorage-in-debugger/#session-and-local-storage

**Comment by mattwojo** [comment]
@dstorey I'm guessing this is all connected to the broken anchor links issue...

http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/

Broken fragments:
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#layout (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#color-picker (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#changes (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#events (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#computed (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#styles (line 49)
http://dev.modern.ie/platform/documentation/f12-devtools-guide/dom-explorer/#pseudo-state (line 49)

**Comment by mattwojo** [comment]
@dstorey -- the two main things from this that I need clarification on are:
1) "ID doesn’t have hyphens"
2) " link lacking its trailing /"

I believe that the other issues raised have either been resolved or will be resolved by  issue #14. 

*In the future, please log issues separately so that they can be individually closed as resolved. ...but thank you for reviewing these docs!!! Very appreciated!!!

**Comment by dstorey** [comment]
ID doesn't have hyphens:

The IDs you use on the named anchors don't use hyphens, so http://dev.modern.ie/platform/documentation/f12-devtools-guide/debugger/#setting-and-managing-breakpoints  fails. Should be #settingandmanagingbreakpoints . I believe that name attribute exists

Link lacking trailing slash:

If you link to something like foo.com/bar, often it will redirect to foo.com/bar/ . If you include the trailing slash in the link, it will avoid that redirect and load faster. i.e. the MathML link should end MathML/

**Comment by mattwojo** [comment]
Aha, I see what you mean about the hyphens now -- thanks @dstorey.  I'm moving this to issue https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/issues/14

**Comment by mattwojo** [comment]
The trailing slash thing is interesting! I think I see what you mean about the link redirect @dstorey. That seems like strange behavior and I don't think most sites do that...? 
Here's a couple of things I found from StackOverflow and Google's Dev / SEO site on the topic. 

Trailing slash in URLs - which style is preferred?
http://stackoverflow.com/questions/5948659/trailing-slash-in-urls-which-style-is-preferred 

To slash or not to slash
http://googlewebmastercentral.blogspot.fr/2010/04/to-slash-or-not-to-slash.html?m=1

I've heard the trailing slash implies it's a directory and not slash means a file, but have never paid much attention before. I've changed that w3c URL to include the trailing / to avoid redirect, though generally it seems that most folks prefer no slash.  I wonder what is behind w3c using a redirect?
